File: sdks/php/src/Exception/ATAPException.php
<?php

declare(strict_types=1);

namespace Atap\Sdk\Exception;

/**
 * Base exception for all ATAP SDK errors.
 */
class ATAPException extends \RuntimeException
{
    public function __construct(
        string $message = '',
        public readonly int $statusCode = 0,
        ?\Throwable $previous = null,
    ) {
        parent::__construct($message, $statusCode, $previous);
    }
}

File: sdks/java/src/main/java/dev/atap/sdk/exception/ATAPException.java
package dev.atap.sdk.exception;

/**
 * Base exception for all ATAP SDK errors.
 */
public class ATAPException extends RuntimeException {

    private final int statusCode;

    public ATAPException(String message) {
        this(message, 0);
    }

    public ATAPException(String message, int statusCode) {
        super(message);
        this.statusCode = statusCode;
    }

    public ATAPException(String message, int statusCode, Throwable cause) {
        super(message, cause);
        this.statusCode = statusCode;
    }

    public int getStatusCode() {
        return statusCode;
    }
}
